universal-application-tool-0.0.1/app/views/applicant/ApplicantProgramSummaryView.java

```diff
@@ -152,9 +158,14 @@ private ContainerTag renderQuestionSummary(
 
     ContainerTag answerContent;
     if (data.fileKey().isPresent()) {
-      String fileLink =
-          controllers.routes.FileController.show(applicantId, data.fileKey().get()).url();
-      answerContent = a().withHref(fileLink).withClasses(Styles.W_2_3);
+      try {
+        String encodedUrl = URLEncoder.encode(data.fileKey().get(), StandardCharsets.UTF_8);
+        String fileLink = controllers.routes.FileController.show(applicantId, encodedUrl).url();
```

Comment:
question (blocking): Would this encoded string need to be decoded in the FileController.show method?

Reply:
Done.